### PR TITLE
Feature/app migrate to hooks and typescript

### PR DESCRIPTION
**Pull request by hacheG**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+interface LoadComponentTypes {
+  logoSet: keyof LogosConfig | null;
+  name: string;
+  component: React.ReactNode;
+  className: string;
+}
+
+const App: React.FunctionComponent = () => {
+  const [user, setUser] = useState<null>(null);
```
[the remark this answers is not in the document]
Comment (erikasv) [reply src/App.tsx:30]: Also, instead of just null in the type of useState, set the newly defined type as an option

File: src/App.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+  const location = useLocation();
+
+  useEffect(() => {
+    isFlagEnabled("CBMModule").then((value) => setShowCBMDashboard(value));
+  }, []);
+
+  const buildQuery = (queryString: string) => new URLSearchParams(queryString);
+
+  const setUserFunc = (user: null | Response | any): void => setUser(user);
```
Comment (erikasv) [inline src/App.tsx:42]: I don't think this function is necessary, try removing it and passing `setUser` directly to Uim

File: src/App.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+      headerNames: { parent, child },
+    });
+  };
+
+  const loadHome = () =>
+    loadComponent({
+      logoSet: "default",
+      name: "",
+      component: <Home referrer={location.pathname} />,
+      className: "fullgrid",
```
Comment (erikasv) [inline src/App.tsx:55]: Home didn't have this class, why did you added it?

**Review by erikasv — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+  const location = useLocation();
+
+  useEffect(() => {
+    isFlagEnabled("CBMModule").then((value) => setShowCBMDashboard(value));
+  }, []);
+
+  const buildQuery = (queryString: string) => new URLSearchParams(queryString);
+
+  const setUserFunc = (user: UserTypes | null) => setUser(user);
```
Comment (erikasv) [inline src/App.tsx:44]: I did the change to pass directly setUser and it works. This function is unnecessary.

File: src/types/loginUimProps.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+export interface CompanyTypes {
```
Comment (erikasv) [inline src/types/loginUimProps.ts:1]: don't export this type

**Review by erikasv — CHANGES_REQUESTED** [review]

**Review by ManuelStardust — APPROVED** [review]

**Review by cazapatamar — APPROVED** [review]

**Pull request merged**